pkg/ledger/ledger.go

```diff
@@ -52,46 +52,82 @@ func (l *Ledger) Close(ctx context.Context) error {
 	return nil
 }
 
-func (l *Ledger) Commit(ctx context.Context, ts []core.Transaction) ([]core.Transaction, error) {
+type Balances map[string]map[string]int64
+
+type CommitTransactionResult struct {
+	core.Transaction
+	Err *TransactionCommitError `json:",omitempty"`
+}
+
+func (l *Ledger) Commit(ctx context.Context, ts []core.Transaction) (Balances, []CommitTransactionResult, error) {
 	unlock, err := l.locker.Lock(l.name)
 	if err != nil {
-		return nil, errors.Wrap(err, "unable to acquire lock")
+		return nil, nil, errors.Wrap(err, "unable to acquire lock")
 	}
 	defer unlock()
 
 	count, _ := l.store.CountTransactions(ctx)
-	rf := map[string]map[string]int64{}
 	timestamp := time.Now().Format(time.RFC3339)
 
 	last, err := l.store.LastTransaction(ctx)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	for i := range ts {
+	mapping, err := l.store.LoadMapping(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
 
-		if len(ts[i].Postings) == 0 {
-			return ts, NewValidationError("transaction has no postings")
-		}
+	contracts := make([]core.Contract, 0)
+	if mapping != nil {
+		contracts = append(contracts, mapping.Contracts...)
+	}
+	contracts = append(contracts, DefaultContracts...)
+
+	aggregatedBalances := make(map[string]map[string]int64)
```

Comment:
Shouldn't we use the balances type here?

Reply:
Oh yes, these syntax are equivalent.